owncloudApp/src/main/java/com/owncloud/android/providers/DocumentsStorageProvider.kt

```diff
@@ -271,27 +264,24 @@ class DocumentsStorageProvider : DocumentsProvider() {
 
     private fun findFiles(root: OCFile, query: String): Vector<OCFile> {
         val result = Vector<OCFile>()
-        for (f in currentStorageManager!!.getFolderContent(root,false)) {
-            if (f.isFolder) {
-                result.addAll(findFiles(f, query))
-            } else {
-                if (f.fileName.contains(query)) {
-                    result.add(f)
-                }
+
+        val folderContent = currentStorageManager?.getFolderContent(root, false) ?: return result
+
+        for (f in folderContent) {
+            if (f.fileName.contains(query)) {
+                result.add(f)
+                if (f.isFolder) result.addAll(findFiles(f, query))
```

Comment:
Are you sure this is the same logic than before? If f is folder, we are adding it as well as add all while with the previous implementation we just added all.

Reply:
It is not. This way, we can look for not only files but also folders. I think it is better, what do you think?

Reply:
Yes, understood.